src/movies.js

```diff
@@ -2,14 +2,124 @@
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+
+
+function orderByYear(arr) {
+
+
+    const alpha = [...arr].sort(function compare(a, b) {
+        if (a.title < b.title) {
+            return -1;
+          }
+          if (a.title > b.title) {
+            return 1;
+          }
+          return 0;
+    });
+    
+    const byYear = alpha.sort(function compare(a, b) {
+        return a.year - b.year;
+    });
+
+    return byYear;
+
+};
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+
+    const steveMovies = arr.filter(function(movie) {
+        return movie.director === 'Steven Spielberg' && movie.genre.includes("Drama")
+    })
+    return steveMovies.length;
+};
+
+
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically (arr) {
+    const alpha = [...arr].map(function(movie){
+        return movie.title;
+    }).sort();
+    return alpha.slice(0, 20);
+
+};
+
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(arr) {
+
+    let newArr = JSON.parse(JSON.stringify(arr));
+
+    if (newArr.length === 0) return 0;
+
+    newArr = newArr.map(function (movie){
+    if (!movie.rate) movie.rate = 0;
+    return movie;
+    });
+    
+    const sumRating = newArr.reduce(function(sum, movie){
+        return sum + movie.rate;
+    }, 0)
+
+    const avgRating = sumRating / newArr.length;
+
+    return parseFloat(avgRating.toFixed(2));
+};
+
+
 // Iteration 5: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr){
+    
+
+    const dramaMovies = arr.filter(function(movie){
+        return movie.genre.includes("Drama")
+    })
+
+
+    if (dramaMovies.length === 0) return 0;
+
+    const dramaSum = dramaMovies.reduce(function(sum, movie){
+        return sum + movie.rate;
+    }, 0)
+
+    return Math.round((dramaSum / dramaMovies.length) * 100) / 100
```

Comment:
nicely done. You could have also do
`return Number((dramaSum / dramaMovies.length).toFixed(2));`